Sources/HTTP/FoundationClient/Client+Foundation.swift

```diff
@@ -0,0 +1,106 @@
+import Transport
+import Foundation
+import URI
+import Core
+
+/**
+    This is here because it's a protocol requirement that we can't change now. 
+    It allows us to satisfy Client goals and allow FoundationClient to function 
+    using URLSession
+*/
+class FauxStream: Transport.Stream {
+    func setTimeout(_ timeout: Double) throws {
+        fatalError("\(#function) not implemented")
```

Comment:
If someone used Foundation Client then saw "Foo not implemented" it would be hard to trace down he cause. Maybe it would be better to say "Foo not implemented for Foundation client, use a client that supports stream operations."

You could also put the stream inside of the `FoundationClient` class and make it `private StreamShim`, that way no one can access it.